curricula/tests/test_curricula_rendering.py

```diff
@@ -87,30 +87,53 @@ def test_render_lesson(self):
         response = self.client.get('/pl-curriculum/pl-unit/1/')
         self.assertEqual(response.status_code, 200)
 
-    def test_lesson_admin_menu(self):
-        response = self.client.get('/test-curriculum/test-unit/1/')
-        self.assertEqual(response.status_code, 200)
-        self.assertIn('admin_edit', response.content)
-        self.assertNotIn('deepSpaceCopy', response.content)
-        response = self.client.get('/test-curriculum/hoc-unit/1/')
-        self.assertEqual(response.status_code, 200)
-        self.assertIn('admin_edit', response.content)
-        self.assertNotIn('deepSpaceCopy', response.content)
+    def test_render_curriculum_admin_menu(self):
+        self.assert_admin_menu('/test-curriculum/', False)
+        self.assert_admin_menu('/pl-curriculum/', False)
 
-        permission = Permission.objects.get(codename='change_lesson')
-        self.user.user_permissions.add(permission)
-        permission = Permission.objects.get(codename='access_all_lessons')
-        self.user.user_permissions.add(permission)
+        self.add_permission(self.user, 'change_curriculum')
+        self.add_permission(self.user, 'access_all_curricula')
 
         # Copy button appears in admin menu for users with sufficient permissions
-        response = self.client.get('/test-curriculum/test-unit/1/')
-        self.assertEqual(response.status_code, 200)
-        self.assertIn('admin_edit', response.content)
-        self.assertIn('deepSpaceCopy', response.content)
-        response = self.client.get('/test-curriculum/hoc-unit/1/')
+        self.assert_admin_menu('/test-curriculum/', True)
+        self.assert_admin_menu('/pl-curriculum/', True)
+
+    def test_render_unit_admin_menu(self):
+        self.assert_admin_menu('/test-curriculum/test-unit/', False)
+        self.assert_admin_menu('/pl-curriculum/pl-unit/', False)
+
+        self.add_permission(self.user, 'change_unit')
+        self.add_permission(self.user, 'access_all_units')
+
+        # Copy button appears in admin menu for users with sufficient permissions
+        self.assert_admin_menu('/test-curriculum/test-unit/', True)
+        self.assert_admin_menu('/pl-curriculum/pl-unit/', True)
+
+    def test_render_lesson_admin_menu(self):
+        self.assert_admin_menu('/test-curriculum/test-unit/1/', False)
+        self.assert_admin_menu('/test-curriculum/hoc-unit/1/', False)
+        self.assert_admin_menu('/pl-curriculum/pl-unit/1/', False)
+
+        self.add_permission(self.user, 'change_lesson')
+        self.add_permission(self.user, 'access_all_lessons')
```

Comment:
Why do you need both change lesson and access all lessons? I thought access_all_lessons would give you the ability to change Lesson

Reply:
https://github.com/code-dot-org/curriculumbuilder/blob/1f4ed9f0990ac22ab7d567bec6c4ad0a86299bab/curricula/views.py#L258-L259
https://github.com/code-dot-org/curriculumbuilder/blob/1f4ed9f0990ac22ab7d567bec6c4ad0a86299bab/lessons/models.py#L450-L452
https://github.com/code-dot-org/curriculumbuilder/blob/1f4ed9f0990ac22ab7d567bec6c4ad0a86299bab/lessons/models.py#L406-L408

The way we have it set up right now, access_all_lessons is not sufficient.